src/textual/widgets/_tabs.py

```diff
@@ -0,0 +1,440 @@
+from __future__ import annotations
+
+from typing import ClassVar
+
+import rich.repr
+from rich.style import Style
+from rich.text import Text, TextType
+
+from .. import events
+from ..app import ComposeResult, RenderResult
+from ..binding import Binding, BindingType
+from ..containers import Container, Horizontal, Vertical
+from ..css.query import NoMatches
+from ..geometry import Offset
+from ..message import Message
+from ..reactive import reactive
+from ..renderables.underline_bar import UnderlineBar
+from ..widget import Widget
+from ..widgets import Static
+
+
+class Underline(Widget):
```

Comment:
I find this to be a really cool widget. Is it worth documenting it so that others can use it for other things?
Or is this really a `Tabs` thing, in which case we may want to change the name to `_Underline`, or `TabUnderline`, or a similar thing?

Reply:
It is tightly coupled with the tabs widget ATM. It is essentially private since its in `_tabs.py` and no exported anywhere.

Maybe at some point it can be its own thing.